implementations/rust/kex/src/x3dh.rs

```diff
@@ -231,8 +236,8 @@ impl KeyExchanger for X3dhResponder {
                 self.prologue()?;
                 let mut vault = self.vault.lock().unwrap();
                 let signed_prekey = vault.secret_public_key_get(self.signed_prekey)?;
-                let signature = vault.sign(self.identity_key, signed_prekey.as_ref())?;
-                let identity_key = vault.secret_public_key_get(self.identity_key)?;
+                let signature = vault.sign(self.identity_key.unwrap(), signed_prekey.as_ref())?;
```

Comment:
does this need to be handled rather than unwrapping?

Reply:
See above.